### PR TITLE
Refactoring cleanup and add more info to registration secret

### PR DESCRIPTION
**Pull request by pfmackin**
# Description

1. Build the API Server external URL using the kubeadmin configmap contents
2. Minor refactoring and unit tests for K8s code
3. Minor cluster API field name change
4. Add managed cluster name to the registration secret
5. Add minor integration test check for the secret contents
6. Minor name function changes from reconcileXyz to syncXyz as discussed with team

Contributes to VZ-2124

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/clusters/vmc_controller_test.go [hunk 1]
```diff
@@ -112,9 +123,6 @@ func TestCreateVMC(t *testing.T) {
 	mock.EXPECT().
```
Comment (desagar) [inline platform-operator/controllers/clusters/vmc_controller_test.go:123]: comment above this line says "Expect a call to get the Secret with the service account token, return one with the token set" but this code has been changed to return nil.

File: platform-operator/internal/config/config_test.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ import (
 
 // TestConfigDefaults tests the config default values
 // GIVEN a new OperatorConfig object
-//  WHEN I call New
+//  WHEN I call Build
```
Comment (desagar) [inline platform-operator/internal/config/config_test.go:14]: nothing has changed in this code except for the comment - and the method is calling Get, not New or Build

File: platform-operator/internal/k8s/kubeadminconfig.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+// ApiEndpoint contains the kubeadmin config information needed to access the API server
+type ApiEndpoint struct {
+	AdvertiseAddress string `json:"advertiseAddress"`
+	BindPort         string `json:"bindPort"`
+}
+
+const (
+	kubeSystem       = "kube-system"
```
Comment (desagar) [inline platform-operator/internal/k8s/kubeadminconfig.go:27]: these same constants are declared in multiple files - I think they should be publicly declared in one location (such as here in kubeadminconfig.go) and referenced in other places.

File: platform-operator/controllers/clusters/vmc_controller_test.go (hunk 1 above)
Comment (pfmackin) [reply platform-operator/controllers/clusters/vmc_controller_test.go:123]: fixed

File: platform-operator/internal/config/config_test.go (hunk 2 above)
Comment (pfmackin) [reply platform-operator/internal/config/config_test.go]: fixed

**Review by mgianatagh — DISMISSED** [review]

**Review by mgianatagh — APPROVED** [review]

**Review by desagar — APPROVED** [review]

**Pull request merged**